packages/rxtranscribe/src/operators/toGCP.js

```diff
@@ -0,0 +1,65 @@
+import { of, from, throwError, combineLatest, forkJoin, zip} from 'rxjs';
+import { mergeMap, concatMap, takeUntil, map, tap, merge, switchMap, bufferCount, bufferTime, mergeAll, concatAll} from 'rxjs/operators';
+import gcpSpeech from '@google-cloud/speech';
+// import bufferBetweenSilences from '../operators/bufferBetweenSilences';
+
+const toGCP = function toGCP({
+  googleCreds = process.env.GOOGLE_APPLICATION_CREDENTIALS,
+  stop$ = of(),
+  sampleRate
+}) {
+  return fileChunk$ => {
+    // Creates a client
+    const clientv1 = new gcpSpeech.SpeechClient();
+    // const clientv1p1beta1 = new gcpSpeech.v1p1beta1.SpeechClient();
+    const config = {
+      encoding: 'LINEAR16',
+      sampleRateHertz: sampleRate,
+      languageCode: 'en-US',
+      // audioChannelCount: 2,
+      // enableSeparateRecognitionPerChannel: true,
+    };
+    return fileChunk$.pipe(
+      mergeMap(fileChunk => (
+        !googleCreds
+        ? throwError(new Error('Google Application Credentials must be set'))
+        : of(fileChunk)
+      )),
+      bufferCount(3),
+      // bufferBetweenSilences({timeInterval: 1000, vadOptions: { sampleRate }}),
+      mergeAll(),
```

Comment:
what does `mergeAll` do here?  I thought it was for flattening an observable of observables into a single stream but the output of the previous step seems to be an array (each of 3 file chunks)?

Reply:
`bufferCount(3)` creates an array of 3 chunks (each arrays  of bytes) and then `mergeAll()` takes that array of arrays and flattens it creating a chunk that is three times the length of the original to be sent through the engine.

Reply:
Further down in the commented out code you can see how I'm using `MergeMap` with `ForkJoin` to resolve an observable of observables. I was sad I couldn't get the two different clients working simultaneously there...